sympy/printing/str.py

```diff
@@ -74,13 +74,20 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~%s' %(self.parenthesize(expr.args[0],PRECEDENCE["Not"]))
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return self.stringify(expr.args, " & ", 38)
+    # The number 38 has been used (instead of PRECEDENCE["And"]) as
```

Comment:
I think this needs some more meditation. Different printers have different precedences, for example the C and Fortran printers would expect `And` to have lower priority than inequalities.

I once create `precedence_traditional` for the LaTeX and pretty printers, maybe we should have something like `precedence_str`.